### ISSUE TITLE
Delete collection for chroma not Working.

### ISSUE THREAD

**Issue by rabin3030** [langchain-ai/langchain#24650]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```
class VectorStoreCreator:
    """
    A class to create a vector store from documents.

    Methods
    -------
    create_vectorstore(documents, embed_model, filepath):
        Creates a vector store from a set of documents using the provided embedding model.
    """
    
    @staticmethod
    def create_vectorstore(documents, embed_model, collection_name):
        """
        Creates a vector store from a set of documents using the provided embedding model.

        This function utilizes the Chroma library to create a vector store, which is a 
        data structure that facilitates efficient similarity searches over the document 
        embeddings. Optionally, a persistent directory and collection name can be specified 
        for storing the vector store on disk.

        Parameters
        ----------
        documents : list
            A list of documents to be embedded and stored.
        embed_model : object
            The embedding model used to convert documents into embeddings.
        filepath : str
            The file path for persisting the vector store.

        Returns
        -------
        object
            A Chroma vector store instance containing the document embeddings.
        """
        try:
            # Create the vector store using Chroma
            vectorstore = Chroma.from_texts(
                texts=documents,
                embedding=embed_model,
                # persist_directory=f"chroma_db_{filepath}",
                collection_name=f"{collection_name}"
            )
            logger.info("Vector store created successfully.")
            return vectorstore
        except Exception as e:
            logger.error(f"An error occurred during vector store creation: {str(e)}")
            return None
    
    @staticmethod
    def create_collection(file_name):
        """
        Create a sanitized collection name from the given file name.
        
        This method removes non-alphanumeric characters from the file name and truncates it to a maximum of 36 characters to form the collection name.
        
        Args:
            file_name (str): The name of the file from which to create the collection name.

        Returns:
            str: The sanitized and truncated collection name.

        Raises:
            Exception: If an error occurs during the collection name creation process, it logs the error.
        """
        try:
            collection_name = re.compile(r'[^a-zA-Z0-9]').sub('', file_name)[:36]
            logger.info(f"A collection name created for the filename: {file_name} as {collection_name}")
            return collection_name
        except Exception as e:
            logger.error(f"An errro occured during the collection name creation : {str(e)}")
        
    @staticmethod
    def delete_vectorstore(collection_name):
        """
        Delete the specified vector store collection.
        
        This method deletes a collection in the vector store identified by the collection name.
        
        Args:
            collection_name (str): The name of the collection to delete.

        Returns:
            None: This method does not return a value.

        Raises:
            Exception: If an error occurs during the deletion process, it logs the error.
        """
        try:
            Chroma.delete_collection()
            return None
        except Exception as e:
            logger.error(f"An error occured during vector store deletion:{str(e)}")
            return None
        ```

### Description

I am trying to delete the collection while using the chroma. But actually it's not working. Could anyone help me to fix this issues.
```
class VectorStoreCreator:
    """
    A class to create a vector store from documents.

    Methods
    -------
    create_vectorstore(documents, embed_model, filepath):
        Creates a vector store from a set of documents using the provided embedding model.
    """
    
    @staticmethod
    def create_vectorstore(documents, embed_model, collection_name):
        """
        Creates a vector store from a set of documents using the provided embedding model.

        This function utilizes the Chroma library to create a vector store, which is a 
        data structure that facilitates efficient similarity searches over the document 
        embeddings. Optionally, a persistent directory and collection name can be specified 
        for storing the vector store on disk.

        Parameters
        ----------
        documents : list
            A list of documents to be embedded and stored.
        embed_model : object
            The embedding model used to convert documents into embeddings.
        filepath : str
            The file path for persisting the vector store.

        Returns
        -------
        object
            A Chroma vector store instance containing the document embeddings.
        """
        try:
            # Create the vector store using Chroma
            vectorstore = Chroma.from_texts(
                texts=documents,
                embedding=embed_model,
                # persist_directory=f"chroma_db_{filepath}",
                collection_name=f"{collection_name}"
            )
            logger.info("Vector store created successfully.")
            return vectorstore
        except Exception as e:
            logger.error(f"An error occurred during vector store creation: {str(e)}")
            return None
    
    @staticmethod
    def create_collection(file_name):
        """
        Create a sanitized collection name from the given file name.
        
        This method removes non-alphanumeric characters from the file name and truncates it to a maximum of 36 characters to form the collection name.
        
        Args:
            file_name (str): The name of the file from which to create the collection name.

        Returns:
            str: The sanitized and truncated collection name.

        Raises:
            Exception: If an error occurs during the collection name creation process, it logs the error.
        """
        try:
            collection_name = re.compile(r'[^a-zA-Z0-9]').sub('', file_name)[:36]
            logger.info(f"A collection name created for the filename: {file_name} as {collection_name}")
            return collection_name
        except Exception as e:
            logger.error(f"An errro occured during the collection name creation : {str(e)}")
        
    @staticmethod
    def delete_vectorstore(collection_name):
        """
        Delete the specified vector store collection.
        
        This method deletes a collection in the vector store identified by the collection name.
        
        Args:
            collection_name (str): The name of the collection to delete.

        Returns:
            None: This method does not return a value.

        Raises:
            Exception: If an error occurs during the deletion process, it logs the error.
        """
        try:
            Chroma.delete_collection()
            return None
        except Exception as e:
            logger.error(f"An error occured during vector store deletion:{str(e)}")
            return None

```

### System Info

langchain==0.1.10

**Comment by jczic** [comment]
Hello @rabin3030,

I had the same problem in production and it was very serious for our company!

We add collections with many vectors/documents and update them very often.
The problem is that if you take a closer look at the SQLite3 database, all the deleted information with deleted links (foreign keys) keeps adding up, but the DB keeps getting bigger and bigger.
In a short space of time we reached over 13 GB in the ChromaDB database folder and the server memory was exploding!

I found a strange, and temporary, solution by testing numerous solutions....
Here it is:

```python
            ids = chromaColl.get()['ids']
            if ids :
                chromaColl.delete(ids)
            del chromaColl
            _chromadb.delete_collection(collectionName)
```

Why is it absolutely necessary to call these 2 deletions in order to empty the data correctly?
